server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -1100,12 +1116,33 @@ export function configureWebSocketServices(
 							: [contentBatch];
 
 						for (const content of contents) {
-							const signalMessage: ISignalMessage = {
-								clientId,
-								content,
-							};
+							let signalMessage: ISignalMessage;
+							let roomId: string;
+							const feature_submit_signals_v2 = "submit_signals_v2";
+
+							if (
+								supportedFeaturesMap.get(clientId)?.clientSupportedFeatures?.[
+									feature_submit_signals_v2
+								]
+							) {
+								signalMessage = {
+									clientId,
+									content: content.content,
+									targetClientId: content.targetClientId,
```

Comment:
This structure does not look the same to me as 1138.
I am not sure typing is correct throughout.
Does an `IDocumentMessage` ever get here?

Reply:
IDocumentMessage shouldn't get here since submit_signal_v2 signals of type ISentSignalMessage. Similar difference in structure is also seen in ODSP service: https://onedrive.visualstudio.com/OneDrive%20Service/_git/PushChannel?path=/src/server/roles/fluid/frontend/fluidWebSocketServer.ts&version=GBmaster&line=1232&lineEnd=1273&lineStartColumn=1&lineEndColumn=6&lineStyle=plain&_a=contents

Reply:
Just added a stronger check just in case, to make sure IDocumentMessage does not get there.